sycl/test/esimd/enums.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clangxx -fsycl -fsyntax-only -Xclang -verify %s
+// RUN: %clangxx -fsycl -fsycl-device-only -fsyntax-only -Xclang -verify %s
```

Comment:
Copying discussion from #3777 here:

@DenisBakhvalov:

> We still want to check host compilation of ESIMD APIS.
> I found some explanation here 7669de4:
>
> > The reason for the change is that "-verify" mode does not work correctly
> > with new integration footer approach, because it seems to ignore line
> > markers in the input file.
>
> But can you please elaborate on it? To me it seems that we silently hide some problem in host compilation.

@AlexeySachkov:

> >But can you please elaborate on it?
>
> Sure, I plan to outline those diagnostics tests changes into a separate PR with more detailed commit message - stay tuned :)
>
> > To me it seems that we silently hide some problem in host compilation.
>
> Hopefully, we don't have different API endpoints for host and device APIs. BTW, is ESIMD classes like simd are expected to work in host part of the program?

@DenisBakhvalov:

> > Hopefully, we don't have different API endpoints for host and device APIs.
>
> Actually, for ESIMD, we do :) . device APIs are lowered into GenX intrinsics, while for host we have an impl using simple loops. That's why we still want to check host compilation, because it's two different paths.
>
> > BTW, is ESIMD classes like simd are expected to work in host part of the program?
>
> No, but they are expected to work in the device part of the program executed on the host.

Reply:
@DenisBakhvalov 

> Actually, for ESIMD, we do :) . device APIs are lowered into GenX intrinsics, while for host we have an impl using simple loops. That's why we still want to check host compilation, because it's two different paths.

It is ok for them to have different implementation between host and device and this is true for like almost all device APIs in our implementation, but I was talking more about the interface itself: I would expect it to be the same between host and device, i.e. same arguments, same names, same return values, same deprecation rules, etc. If that is not the case, then I think that something is wrong with design of the extension.

Please also take a look at the commit message: there are couple of other points related to this too

Reply:
I'm not sure I understand how different implementation and design of the extension is relevant here.
For ESIMD we need to test host compilation even if it does not run on host device. So please don't add `-fsycl-device-only` for ESIMD tests.
I wonder why this is not the case for SYCL - same applies. Host compiler and device compiler use different code to compile - e.g. due to __SYCL_DEVICE__ONLY__, so having code compiled by device compiler does not mean it can be compiled by the host compiler.

Reply:
This particular test checks that we have a warning about a method (or something else) being deprecated, it does not check full compilation or execution. Do we really expect to have an API which is deprecated on host, but not deprecated on device or vice versa?

Reply:
I've read description of those tests again and indeed they were intended to not only check deprecation messages, but rather check that we can compile particular code, in which case host compilation must also be checked.

Unfortunately, we won't be able to use `-vefiry` for that, but it should be possible to workaround that with `FileCheck`, I will prepare an update for this PR

Reply:
My main point above is that compilation should be checked for both host and device compiler, whether it is SYCL or ESIMD. Do you agree with that?

Regarding this particular test - it does check compilation, and this is one of its purposes, as well as checking the deprecation note. I'm not sure what do you mean by "full".

Reply:
> My main point above is that compilation should be checked for both host and device compiler, whether it is SYCL or ESIMD. Do you agree with that?

I still think that it depends on the particular test. For example some tests which I've modified do not even have device code in them and if the only thing we care is some diagnostic messages, then we could probably optimize them by avoiding unnecessary compiler passes.

>  I'm not sure what do you mean by "full".

Something closer to E2E test: modified tests contain `-fsyntax-only`, which means that we stop right after FE, I'm not even sure that LLVM IR is generated by clang in that case.

Reply:
For tests w/o device code - maybe. Even though this might not be reliable too as device compiler may use different includes and can fail parsing/syntax check.

We don't need to generate IR in such tests, just check that API compiles, as this is not E2E tests.

Reply:
> We don't need to generate IR in such tests, just check that API compiles, as this is not E2E tests.

unless, of course, the purpose of the test is to check FE CG. There are some cases like these.

Reply:
I've updated the PR to enable host compilation back for ESIMD tests in 118ec382ffcb5d04ae1bf317e705024f875591bd, but I had to use `FileCheck`, because `-verify` won't work for host part in foreseeable feature